### DISCUSSION TITLE
Running the counter example on a GCP VM

### DISCUSSION THREAD

**Discussion by sanket28** [cilium/ebpf#1480] [Q&A]
Hello,

I'm trying to run the counter example from https://ebpf-go.dev/guides/getting-started/ on a GCP VM. 
Here are the details:
```
Linux bob-1 5.10.0-29-cloud-amd64 #1 SMP Debian 5.10.216-1 (2024-05-03) x86_64 GNU/Linux
```

Here is the output of ifconfig:
```
ens4: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1460
        inet 10.128.0.65  netmask 255.255.255.255  broadcast 10.128.0.65
        inet6 fe80::4001:aff:fe80:41  prefixlen 64  scopeid 0x20<link>
        ether 42:01:0a:80:00:41  txqueuelen 1000  (Ethernet)
        RX packets 14141  bytes 250349607 (238.7 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 7801  bytes 783668 (765.3 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 20  bytes 1548 (1.5 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 20  bytes 1548 (1.5 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

I'm trying to run the example on the ens4 interface but I get this error:
```
2024/06/06 03:42:38 Attaching XDP:create link: invalid argument
```

Any help would be really appreciated. 

Thanks!

**Comment by lmb** [comment]
Can you strace the example and provide the output here?

**Reply by mmynk** [reply]
@lmb 
I am having the same issue while trying to run the example for `eth0`:
```sh
$ ifconfig | grep mtu
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 3840
lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
```
[bpf_counter.strace.txt](https://github.com/user-attachments/files/16273841/bpf_counter.strace.txt)

**Reply by ti-mo** [reply]
I don't understand, did you modify this strace output? It looks incomplete. For `create link: invalid argument` to show up, there needs to be a failed `bpf(BPF_LINK_CREATE, ...` call, but it's nowhere to be found. Are you running this with `strace -f`?
